util/stringutil/string_util_test.go

```diff
@@ -152,3 +152,21 @@ func (s *testStringUtilSuite) TestIsExactMatch(c *C) {
 		c.Assert(IsExactMatch(patTypes), Equals, v.exactMatch, Commentf("%v", v))
 	}
 }
+
+func BenchmarkMatchSpecial(b *testing.B) {
+	var (
+		pattern = `a%a%a%a%a%a%a%a%b`
+		target  = `aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa`
+		escape  = byte('\\')
+	)
+
+	patChars, patTypes := CompilePattern(pattern, escape)
+
+	b.StartTimer()
```

Comment:
```suggestion
	b.ResetTimer()
```
The benchmarking framework starts the timer automatically, see https://golang.org/pkg/testing/#B.StartTimer .
So we should call b.StopTimer at the begin of `BenchmarkMatchSpecial`, or just call `b.ResetTimer`.